spinalcordtoolbox/reports/assets/_assets/js/main.js

```diff
@@ -1,327 +0,0 @@
-    if(rows[i].innerText == selected_row)
-    {
-      rows[i].className="active";
-      rows[i].scrollIntoView();
-    }
-  }
-  hideColumns();
-});
-
-$("#table").on('sort.bs.table', function (e, row) {
-  
-  $('#table').on('post-body.bs.table', function(e,params){
-                  console.log("sort finish");
-                  $('#table').unbind("post-body.bs.table");
-                  hideColumns();
-                  rows=$("table tbody tr");
-                  for(let i=0; i<rows.length; i++)
-                  {
-                    if(rows[i].innerText == selected_row)
-                    {
-                      rows[i].className="active";
-                      rows[i].scrollIntoViewIfNeeded(false);
-                    }
-                  }
-            });
-  
-});
-
-function responseHandler(res) {
-  var n;
-  for(var i = 0; i < res.length; i++) {
-    n = new Date(res[i].moddate);
-    res[i].moddate = n.toLocaleString();
-  }
-  return res;
-}
-
-function set_download_yml_btn_state(marker) {
-  let disabled = true;
-  sct_data.forEach(item => {
-      if (item.qc === marker) {
-        disabled = false;
-      }
-  });
-  if (containsNonLatinCodepoints(marker) === true) {
-    // This converts e.g. '\u2718' -> '2718' with corresponding id='download_yaml_btn_2718'
-    marker = marker.codePointAt(0).toString(16)
-  }
-  document.getElementById("download_yaml_btn_".concat(marker)).disabled = disabled;
-}
-
-function containsNonLatinCodepoints(s) {
-    return /[^\u0000-\u00ff]/.test(s);
-}
-
-function downloadQcStates() {
-  var qcFlags = {};
-  // Fetch all QC flags from the QC column of the table
-  sct_data.forEach(function(item, index) {
-      var uniqueId = item.moddate + '_' + item.fname_in + '_' + item.command;
-      qcFlags[uniqueId+"_qc"] = item.qc;
-      qcFlags[uniqueId+"_rank"] = item.rank;
-  });
-  // Create a blob and trigger a download
-  var dataStr = "data:text/json;charset=utf-8," + encodeURIComponent(JSON.stringify(qcFlags));
-  var downloadAnchorNode = document.createElement('a');
-  downloadAnchorNode.setAttribute("href", dataStr);
-  downloadAnchorNode.setAttribute("download", "qc_flags.json");
-  document.body.appendChild(downloadAnchorNode);
-  downloadAnchorNode.click();
-  downloadAnchorNode.remove();
-};
-
-function loadAndSetQcStates(event) {
-  var file = event.target.files[0];
-  var reader = new FileReader();
-  reader.onload = function(e) {
-      var qcFlags = JSON.parse(e.target.result);
-      for (var key in qcFlags) {
-          if (qcFlags.hasOwnProperty(key)) {
-              localStorage.setItem(key, qcFlags[key]);
-          }
-          updateQcStates();
-      }
-  };
-  reader.readAsText(file);
-}
-
-function updateQcStates() {
-    // Load and set QC state from local storage
-    // TODO: create a function for this code block
-    sct_data.forEach((item, index) => {
-      var uniqueId = sct_data[index].moddate + '_' + sct_data[index].fname_in + '_' + sct_data[index].command;
```

Comment:
It looks like the old QC report used the `moddate`+`fname_in`+`command` as part of a way to try to guarantee a unique ID for the entry. Which I guess makes sense, given that it's path-independent, and it's unlikely for a QC report to be generated to the exact second with the exact filename and command? :thinking:

Reply:
Tried the new UUID approach on Firefox, and everything seems to be working splendidly! I like that we're keeping the whole table now (as well as the metadata), rather than doing these sorts of ad-hoc per-row unique IDs.

Samples (note the old entries for QCs that were cached from other reports I put on my Desktop, aha):

![image](https://github.com/user-attachments/assets/d007f856-922d-44ec-b89e-50d506e8fb7b)

![image](https://github.com/user-attachments/assets/19c0e8ab-dbd7-42f5-9fd0-e39119deda78)